script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/KasScript.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.coolbitx.wallet.signing.scriptlib;
+
+import com.coolbitx.wallet.signing.utils.HexUtil;
+import com.coolbitx.wallet.signing.utils.ScriptArgumentComposer;
+import com.coolbitx.wallet.signing.utils.ScriptAssembler;
+import com.coolbitx.wallet.signing.utils.ScriptData;
+import com.coolbitx.wallet.signing.utils.ScriptData.Buffer;
+import com.google.common.base.Strings;
+
+public class KasScript {
+
+    public static void main(String[] args) {
+        listAll();
+    }
+
+    public static void listAll() {
+        System.out.println("Kas transfer: \n" + getTransferScript(false) + "\n");
+    }
+
+    public static String getAddressScript(ScriptData argOutputScriptPublicKey) {
+        String hrp = "kaspa";
+        String hrpExpand = "";
+        for (int i = 0; i < hrp.length(); i++) {
+            hrpExpand += HexUtil.toHexString((hrp.charAt(i) >> 5) & 7, 1);
+        }
+        hrpExpand += "00";
+        for (int i = 0; i < hrp.length(); i++) {
+            hrpExpand += HexUtil.toHexString(hrp.charAt(i) & 31, 1);
+        }
+        String bech32AddressScript = new ScriptAssembler()
+                .copyString(hrpExpand + "00", Buffer.CACHE2)
+                .baseConvert(argOutputScriptPublicKey, Buffer.CACHE2, 55,
+                        ScriptAssembler.binary32Charset, ScriptAssembler.bitLeftJustify8to5) // (34 * 8) / 5 = 55
```

Comment:
應該是去掉`length` 和OP_CHECKSIG `AC`的長度?
```suggestion
                .baseConvert(argOutputScriptPublicKey, Buffer.CACHE2, 52,
                        ScriptAssembler.binary32Charset, ScriptAssembler.bitLeftJustify8to5) // (32 * 8) / 5 = 52
```

Reply:
有調整過了